app/assets/javascripts/models/ui/reverb_retirement.coffee

```diff
@@ -0,0 +1,49 @@
+#= require util/js.cookies
+#= require util/metrics
+
+ns = @edsc.models.ui
+data = @edsc.models.data
+
+ns.ReverbRetirement = do (ko,
+                          ajax = @edsc.util.xhr.ajax,
+                          config=@edsc.models.data.config) ->
+  class ReverbRetirement
+    constructor: ->
+
+    referrerIsReverb: () =>
+      console.log "Checking referrer: " + document.referrer
+      console.log "Checking referrer against ENV value: " + edsc.config.reverb_url
+      referrer = if document.referrer then document.referrer.match(/:\/\/(.[^/]+)/)[1] else false
+      reverb = if edsc.config.reverb_url then edsc.config.reverb_url.match(/:\/\/(.[^/]+)/)[1] else false
+      return referrer == reverb && referrer
+
+    returnToReverb: (source) =>
+      Cookies.set('ReadyForReverbRetirement', 'false', { expires: 90 })
+      data['type'] = 'reverb_redirect'
+      data['data'] = 'back_to_reverb'
+      data['other_data'] = "{source: '" + source + "'}"
+      ajax
+        data: JSON.parse(JSON.stringify(data))
+        dataType: 'json'
+        url: "/metrics"
+        method: 'post'
+        success: (data) ->
```

Comment:
Just a heads-up, if the controller returns ```:nothing```, the data will be nil here.

Reply:
That's okay; only used for a console log entry.